### DISCUSSION TITLE
Non si scatenano i webhook

### DISCUSSION THREAD

**Discussion by FrancescoAntonelli** [fattureincloud/api#385] [Q&A] [answered]
Buonasera,
  mi sono registrato agli hook in questa maniera

```
public async Task SubscribeEvents(int companyId, string apiBasePath, string accessToken)
{
		Configuration config = new Configuration
		{
			BasePath = apiBasePath,
			// Configure OAuth2 access token for authorization: OAuth2AuthenticationCodeFlow
			AccessToken = accessToken
		};

		var apiInstance = new WebhooksApi(config);
		
		var createWebhooksSubscriptionRequest = new CreateWebhooksSubscriptionRequest(); // CreateWebhooksSubscriptionRequest |  (optional) 
		var eventSubscriptions = new EventType[]
		{
			EventType.IssuedDocumentsCreditNotesCreate,
			EventType.IssuedDocumentsCreditNotesUpdate,
			EventType.IssuedDocumentsCreditNotesDelete,
			EventType.IssuedDocumentsDeliveryNotesDelete,
			EventType.IssuedDocumentsDeliveryNotesCreate,
			EventType.IssuedDocumentsDeliveryNotesUpdate,
			EventType.IssuedDocumentsEInvoicesStatusUpdate,
			EventType.IssuedDocumentsInvoicesCreate,
			EventType.IssuedDocumentsInvoicesDelete,
			EventType.IssuedDocumentsInvoicesUpdate,
			EventType.IssuedDocumentsSelfInvoicesCreate,
			EventType.IssuedDocumentsSelfInvoicesUpdate,
			EventType.IssuedDocumentsSelfInvoicesDelete,
		};
		createWebhooksSubscriptionRequest.Data =
			new WebhooksSubscription(sink: "https://webhook.site/9fc4bbc3-37d1-4c5f-a7e7-a860001151d9/api/events/listner", types: eventSubscriptions.ToList());
		
		try
		{
			// Create a Webhook Subscription 
			CreateWebhooksSubscriptionResponse result = await apiInstance.CreateWebhooksSubscriptionAsync(companyId, createWebhooksSubscriptionRequest);
			Debug.WriteLine(result);
		}
		catch (ApiException e)
		{
			Debug.Print("Exception when calling WebhooksApi.CreateWebhooksSubscription: " + e.Message);
			Debug.Print("Status Code: " + e.ErrorCode);
			Debug.Print(e.StackTrace);
		}
}
```
Tramite Webhook.site riesco ad arrivare alla ricezione del welcome message, ma poi se vado nella mia utenza di FiC e modifico o aggiungo una fattura non si scatena nessun evento. C'è qualcosa che faccio in maniera scorretta o qualche verifica che possa fare?

Grazie per il supporto
   Francesco

**Comment by emajo** [comment, marked as answer]
Ciao @FrancescoAntonelli credo che il problema sia che lei stia provando a registrare tipi di eventi già registrati, il mio consiglio è quello di effettuare manualmente una chiamata [List Webhooks Subscritpions](https://developers.fattureincloud.it/api-reference#get-/c/-company_id-/subscriptions) e vedere se gli eventi interessati sono già registrati ed eventualmente eliminare le subscription non utilizzate. La chiamata di registrazione subscription infatti non ritorna errore se alcuni degli eventi risultano già registrati ma ritorna un array (il campo `warnings`) con gli gli eventi non registrati.

Grazie,
Emanuele.

**Reply by valmoz** [reply]
Ciao @FrancescoAntonelli per completezza aggiungo la parte dedicata a questo argomento sulla documentazione:
![image](https://github.com/fattureincloud/api/assets/7254293/3c1c008e-0acc-412c-9492-ba72df1be107)
Puoi trovare il dettaglio qui: [Subscriptions > Subscribe](https://developers.fattureincloud.it/docs/webhooks/subscriptions#%EF%B8%8F-subscribe)

**Reply by FrancescoAntonelli** [reply]
Grazie, il problema era legato al numero di sottoscrizioni che avevo fatto. Le ho cancellate tutte rifatta una nuova e tutto ha preso a funzionare.

Francesco
